src/main/java/org/kohsuke/github/GHIssue.java

```diff
@@ -411,15 +405,9 @@ public void removeLabels(Collection<GHLabel> labels) throws IOException {
     }
 
     private void _removeLabels(Collection<String> names) throws IOException {
-        List<String> newLabels = new ArrayList<String>();
-
-        for (GHLabel l : getLabels()) {
-            if (!names.contains(l.getName())) {
-                newLabels.add(l.getName());
-            }
+        for (String name : names) {
+            root.createRequest().method("DELETE").withUrlPath(getIssuesApiRoute() + "/labels", name).send();
```

Comment:
Ouch. I didn't notice this behavior. Not great for rate limit budget... But it is correct.  Good job.

Reply:
Yeah, I agree it's unfortunate they did it that way. It should have been symmetrical with adding labels IMHO.